taraxa/state/dpos/precompiled/delegations.go

```diff
@@ -42,6 +42,12 @@ func (self *Delegations) DelegationExists(delegator_address *common.Address, val
 	return delegator_validators.AccountExists(validator_address)
 }
 
+// Returns number of delegations for specified address
+func (self *Delegations) GetDelegationsCount(delegator_address *common.Address) uint32 {
+	delegator_validators := self.getDelegatorValidatorsList(delegator_address)
```

Comment:
I do not think this is right as you are creating new iterable map here in case nothing is found. I think it should be called directly to `delegator_validators, found := self.delegators_validators[*delegator_address]` and call getcount only when it is found

Reply:
I think you dont understand how this structure works. The original approach you implemented was actually a bug. You have to check database every time to get this count. To be able to check database, you have to create IterableMap(that means creating storage keys prefixes) and then check database.